src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PrivateDataRetrievalService.java

```diff
@@ -1,125 +1,242 @@
+                    logMap);
+            throw new ServiceException(
+                    String.format("Non-hashed ID could not be found for Hashed ID: %s",
+                            hashedTrustId));
+        }
+
+        ApiLogger.info("Retrieving Corporate Trustee data for Trust Id" + nonHashedId, logMap);
+
+        try {
+            PrivateCorporateTrusteeListApi corporateTrustees = apiClientService.getInternalApiClient()
+                    .privateCorporateTrusteeDataResourceHandler().getCorporateTrusteeData(
+                            OVERSEAS_ENTITY_URI_SECTION + "trusts/" + nonHashedId
+                                    + "/corporate-trustees").execute().getData();
+
+            if (corporateTrustees != null && corporateTrustees.getData() != null
+                    && !corporateTrustees.getData().isEmpty()) {
+                ApiLogger.info(String.format("Retrieved %d Trusts for Company Number %s",
+                        corporateTrustees.getData().size(), hashedTrustId));
+            }
+
+            return corporateTrustees;
+
+        } catch (ApiErrorResponseException e) {
+            if (e.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
+                ApiLogger.info("No Corporate Trustee found for Trust Id " + hashedTrustId, logMap);
+                return new PrivateCorporateTrusteeListApi(Collections.emptyList());
+            }
+            throw new ServiceException(e.getStatusMessage(), e);
+        } catch (URIValidationException e) {
+            var message = "Error Retrieving Corporate Trustee data for Trust Id " + hashedTrustId;
+            ApiLogger.errorContext(message, e);
+            throw new ServiceException(e.getMessage(), e);
+        }
+
     }
-  }
-
-  public OverseasEntityDataApi getOverseasEntityData(String companyNumber)
-      throws ServiceException {
-    var logMap = new HashMap<String, Object>();
-    try {
-      var overseasEntityDataApi = apiClientService
-          .getInternalApiClient()
-          .privateOverseasEntityDataHandler()
-          .getOverseasEntityData(OVERSEAS_ENTITY_URI_SECTION + companyNumber + "/entity-data")
-          .execute()
-          .getData();
-
-
-      logMap.put(COMPANY_NUMBER, companyNumber);
-      ApiLogger.info("Retrieving overseas entity data for company number ",  logMap);
-
-      return overseasEntityDataApi;
-    } catch (ApiErrorResponseException e) {
-      if (e.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
-        ApiLogger.info("No overseas entity data for Company Number " + companyNumber, logMap);
-        return new OverseasEntityDataApi();
-      }
-      throw new ServiceException(e.getStatusMessage(), e);
-    } catch (URIValidationException e) {
-      var message = "Error Retrieving overseas entity data for " + companyNumber;
-      ApiLogger.error(message, e, logMap);
-      throw new ServiceException(e.getMessage(), e);
+
```

Comment:
assume this (getTrustDetails) will be in other PR

Reply:
If I remove it it won't compile and therefore won't be able to pass the concourse-ci checks.

It might be best to wait till [Barrys PR](https://github.com/companieshouse/overseas-entities-api/pull/346) is merged